website/static/js/accountSettings.js

```diff
@@ -0,0 +1,170 @@
+var oop = require('js/oop.js');
+var Raven = require('raven-js');
+
+require('knockout.punches');
+ko.punches.enableAll();
+
+
+var userEmail = oop.defclass({
+    constructor: function(data) {
+        this.address = ko.observable();
+        this.isConfirmed = ko.observable();
+        this.isPrimary = ko.observable();
+        if (typeof(data) !== 'undefined') {
+            this.unserialize(data);
+        }
+    },
+    serialize: function() {
+        return {
+            address: this.address(),
+            primary: this.isPrimary(),
+            confirmed: this.isConfirmed()
+        };
+    },
+    unserialize: function(data) {
+        this.address(data.address);
+        this.isPrimary(data.primary || false);
+        this.isConfirmed(data.confirmed || false);
+    }
+});
+
+
+var userProfile = oop.defclass({
+    constructor: function(urls) {
+        this.urls = urls;
+
+        // TODO: Pass these in instead!
+        this.urls = {
+            fetch: '/api/v1/profile/',
+            update: '/api/v1/profile/'
+        };
+
+        this.id = ko.observable();
+        this.emails = ko.observableArray();
+
+        this.primaryEmail = ko.computed(function () {
+            var emails = this.emails();
+            for (var i = 0; i < this.emails().length; i++) {
+                if(emails[i].isPrimary()) {
+                    return emails[i];
+                }
+            }
+            return new userEmail();
+        }.bind(this));
+
+        this.alternateEmails = ko.computed(function () {
+            var emails = this.emails();
+            var retval = [];
+            for (var i = 0; i < this.emails().length; i++) {
+                if (emails[i].isConfirmed() && !emails[i].isPrimary()) {
+                    retval.push(emails[i]);
+                }
+            }
+            return retval;
+        }.bind(this));
+
+        this.unconfirmedEmails = ko.computed(function () {
+            var emails = this.emails();
+            var retval = [];
+            for (var i = 0; i < this.emails().length; i++) {
+                if(!emails[i].isConfirmed()) {
+                    retval.push(emails[i]);
+                }
+            }
+            return retval;
+        }.bind(this));
+
+        // user inputs
+        this.emailInput = ko.observable();
+    },
+    init: function () {
+        this.fetchData(this.unserialize.bind(this));
+    },
+    serialize: function () {
+        var emails = ko.utils.arrayMap(this.emails(), function(each) {
+            return each.serialize();
+        });
+        return {
+            id: this.id(),
+            emails: emails
+        };
+    },
+    unserialize: function (data) {
+        var self = this;
+        var profile = data.profile;
+
+        this.id(profile.id);
+        self.emails([]);
+        ko.utils.arrayMap(profile.emails, function(each) {
+            self.emails.push(new userEmail(each));
+        });
+    },
+    fetchData: function (callback) {
```

Comment:
It may make sense to separate data fetching from the ViewModel. Perhaps create a `ProfileClient` that does the work of fetching email info from the server.